File: src/tests/integration/Bc.Endpoint.Integration.Tests/CommentTakingSender.cs
using System;
using System.Threading.Tasks;
using Bc.Contracts.Internals.Endpoint.CommentTaking.Commands;
using NServiceBus;
using NUnit.Framework;

namespace Bc.Endpoint.Integration.Tests
{
    [TestFixture]
    [Ignore("Only for manual tests")]
    public class CommentTakingSender
    {
        private IEndpointInstance endpointInstance;

        [SetUp]
        public async Task SetUp()
        {
            this.endpointInstance = await EndpointFactory.GetSenderEndpoint().ConfigureAwait(false);
        }

        [TearDown]
        public Task TearDown()
        {
            return this.endpointInstance.Stop();
        }

        [Test]
        public async Task TakeComment_Send_NoException()
        {
            // Arrange
            var commentId = Guid.NewGuid();
            const string userName = "test_user";
            const string userEmail = "test_user_email";
            const string userWebsite = "test_user_website";
            const string fileName = @"_posts/2018-05-27-test.md";
            const string content = "new_comment";
            var addedDate = DateTime.UtcNow;

            var message = new TakeComment(commentId, userName, userEmail, userWebsite, fileName, content, addedDate);

            // Act
            await this.endpointInstance.Send(message).ConfigureAwait(false);

            // Assert
            Assert.Pass();
        }
    }
}

File: src/tests/integration/Bc.Endpoint.Integration.Tests/CommentTakingSender.fs
module Bc.Endpoint.Integration.Tests.CommentTakingSender

open System
open System.Threading.Tasks
open Bc.Contracts.Internals.Endpoint.CommentTaking.Commands
open NServiceBus
open NUnit.Framework

[<Ignore("Only for manual tests")>]
module Sender =

    let sendMessage<'T> (message:'T) =
        async {
            let! endpoint = EndpointFactory.getSenderEndpoint () |> Async.AwaitTask
            do! endpoint.Send(message) |> Async.AwaitTask
            do! endpoint.Stop() |> Async.AwaitTask
        } |> Async.RunSynchronously

    [<Test>]
    let takeComment_send_noException () =

        // Arrange
        let commentId = Guid.NewGuid()
        let userName = "test_user"
        let userEmail = "test_user_email"
        let userWebsite = "test_user_website"
        let fileName = @"_posts/2018-05-27-test.md"
        let content = "new_comment"
        let addedDate = DateTime.UtcNow

        let message =
            TakeComment (
                commentId,
                userName,
                userEmail,
                userWebsite,
                fileName,
                content,
                addedDate
            )

        // Act
        sendMessage<TakeComment> message

        // Assert
        Assert.Pass()
